examples/hif2a/generate_star_map.py

```diff
@@ -0,0 +1,264 @@
+
+    Notes
+    -----
+    TODO: mcs_map should be symmetric, so only have to do the upper triangle of this matrix
+    """
+    mcs_s = np.zeros((len(mols), len(mols)), dtype=object)
+
+    for i in range(len(mols)):
+        for j in range(len(mols)):
+            mcs_s[i, j] = mcs_map(mols[i], mols[j])
+    return mcs_s
+
+
+def compute_transformation_size_matrix(mols, mcs_s):
+    N = len(mols)
+    transformation_sizes = np.zeros((N, N))
+    for i in range(N):
+        for j in range(N):
+            n_i = mols[i].GetNumAtoms()
+            n_j = mols[j].GetNumAtoms()
+            n_mcs = mcs_s[i, j].numAtoms
+            transformation_sizes[i, j] = transformation_size(n_i, n_j, n_mcs)
+    return transformation_sizes
+
+
+mcs_s = compute_all_pairs_mcs(mols_with_core_1)
+transformation_sizes = compute_transformation_size_matrix(mols_with_core_1, mcs_s)
+
+
+def identify_hub(transformation_sizes):
+    return int(np.argmin(transformation_sizes.sum(0)))
+
+
+hub_index = identify_hub(transformation_sizes)
+hub = mols_with_core_1[hub_index]
+
+
+def plot_transformation_sizes(transformation_sizes):
+    plt.imshow(transformation_sizes)
+    plt.xlabel('molecule index')
+    plt.ylabel('molecule index')
+    plt.title('"size of transformation"\n$(n_A - n_{MCS}) + (n_B - n_{MCS})$')
+
+    plt.tight_layout()
+
+    plt.colorbar()
+
+
+plot_transformation_sizes(transformation_sizes)
+plt.savefig(path_to_results.joinpath('transformation_sizes.png'), bbox_inches='tight')
+plt.close()
+
+
+# 4. Construct and serialize the relative transformations
+def get_core(mol_a, mol_b, query):
+    """Return np integer array that can be passed to RelativeFreeEnergy constructor
+
+    Parameters
+    ----------
+    mol_a, mol_b, query : RDKit molecules
+
+    Returns
+    -------
+    core : np.ndarray of ints, shape (n_MCS, 2)
+
+    TODO: move this into a utility module or the free energy module
+    """
+
+    # fetch conformer, assumed aligned
+    conf_a = mol_a.GetConformer(0).GetPositions()
+    conf_b = mol_b.GetConformer(0).GetPositions()
+
+    # note that >1 match possible here -- must pick minimum-cost match
+    matches_a = mol_a.GetSubstructMatches(query)
+    matches_b = mol_b.GetSubstructMatches(query)
+    n_a, n_b = len(matches_a), len(matches_b)
+
+    # cost[i, j] = sum_i distance(conf)
+    cost = np.zeros((len(matches_a), len(matches_b)))
+    for i, a in enumerate(matches_a):
+        for j, b in enumerate(matches_b):
+            cost[i, j] = np.linalg.norm(conf_a[np.array(a)] - conf_b[np.array(b)], axis=1).sum()
+
+    # find (i,j) = argmin cost
+    min_i, min_j = np.unravel_index(np.argmin(cost, axis=None), cost.shape)
+    print(f'argmin of {n_a} x {n_b} cost matrix: {(min_i, min_j)} ')
+    # TODO: maybe also print the difference between min(cost) and cost[0,0],
+    #   to see how big of a difference it made to pick the default
+
+    # TODO: is there a way to use the matching from MCS directly?
```

Comment:
Unfortunately no, the MCS code actually constructs a queryMol incrementally now that I think about it. It was designed for the multiple ligand use case (i.e. the FMCS code works on >2 number of ligands).